### ISSUE TITLE
Pre-Rector 1.0 - container slim fit

### ISSUE THREAD

**Issue by TomasVotruba** [rectorphp/rector#8120]
Currently, the Symfony container pulls http-kernel and creates cached and compiled container on every test run. This leads to extreemely slow tests and downgrade of full range of symfony/* packages we don't need. I wrote about this topic here:

* https://tomasvotruba.com/blog/experiment-how-i-replaced-symfony-di-with-laravel-container-in-ecs
* https://tomasvotruba.com/blog/unleash-the-power-of-simplicity-php-cli-app-with-minimal-dependencies

I want to make an experiment on performance speed, memory load and /vendor downgrade size in 2 scenarios:

* current Symfony DI
* lightweight Laravel container

If the later proves well, we can switch to it like we did in ECS last week. That resulted in 4 time faster tests:

https://twitter.com/VotrubaT/status/1683576139049058304

<br>

Let's see how the speed goes in here :)

**Comment by staabm** [comment]
That would be awesome. I remember one of the PRs I did in the past had to workaround slow symfony kernel caching/cache-use. 

Getting rid of it would be great

**Comment by yguedidi** [comment]
Hey @TomasVotruba! Nice initiative!

Maybe you can give Symfony a chance, according to the [component doc](https://symfony.com/doc/current/components/dependency_injection.html) it's possible to build a container by adding services :)
Also, regarding the link to HttpKernel, it seem symfony/http-kernel [is not required](https://packagist.org/packages/symfony/dependency-injection) by symfony/dependency-injection.
From running `composer why symfony/http-kernel` we get `symplify/symplify-kernel 11.1.27 requires symfony/http-kernel (6.1.*)`
Did I missunderstood something?

**Comment by TomasVotruba** [comment]
@staabm Indeed, even if illuminate/container will be too-lazy, I think we can easy cache around it's build.

@yguedidi Guess what I've been doing past 8 years :) Symfony is not suitable for CLI apps.
The http-kernel is not needed, but then you have to handle most of the cache yourself. So you need it or copy it :)
See:

* https://tomasvotruba.com/blog/unleash-the-power-of-simplicity-php-cli-app-with-minimal-dependencies
*https://tomasvotruba.com/blog/experiment-how-i-replaced-symfony-di-with-laravel-container-in-ecs

Any PR to improve the speed is welcomed :+1:

**Comment by samsonasik** [comment]
@TomasVotruba I think this can be closed.
